pkg/runtime/reconciler.go

```diff
@@ -346,6 +346,20 @@ func (r *resourceReconciler) createResource(
 		return nil, err
 	}
 
+	// Once the desired resource is marked as managed (i.e. Finalizer is added
```

Comment:
God idea! I will consider this in the separate PR as advised. 
But in this PR, I did not want to introduce any major change in reconciler by moving the `setResourceManaged` call and the extra API server calls did not seem a lot since this code only runs once during lifecycle of a resource(once during create).